stl/inc/vector

```diff
@@ -440,6 +440,189 @@ constexpr _Ty* _Unfancy_maybe_null(_Ty* _Ptr) noexcept { // do nothing for plain
+            return;
+        }
+#endif // __cpp_lib_constexpr_dynamic_alloc
+
+        const auto& _My_data     = _Target_vec._Mypair._Myval2;
+        const _Pointer& _Myfirst = _My_data._Myfirst;
+        const _Pointer& _Mylast  = _My_data._Mylast;
+        const _Pointer& _Myend   = _My_data._Myend;
+
+        if (!_Old_last) {
+            __sanitizer_annotate_contiguous_container(_Myfirst, _Myend, _Myend, _Mylast);
+        } else if (_Mylast < _Old_last) {
+            __sanitizer_annotate_contiguous_container(_Myfirst, _Myend, _Old_last, _Mylast);
+        }
+    }
+
+private:
+    const _Ty& _Target_vec;
+
+    _Pointer _Old_last;
+};
+
+template <class _Ty>
+class _NODISCARD _Asan_annotater<_Ty, false> {
+private:
+    using _Alloc      = typename _Ty::allocator_type;
+    using _Pointer    = typename _Ty::pointer;
+    using _Size_type  = typename _Ty::size_type;
+    using _Value_type = typename _Ty::value_type;
+
+    const void* _Get_aligned_start() {
+        const auto& _My_data       = _Target_vec._Mypair._Myval2;
+        const _Pointer& _Myfirst   = _My_data._Myfirst;
+        const _Pointer& _Myend     = _My_data._Myend;
+        const _Size_type _Capacity = static_cast<_Size_type>(_Myend - _Myfirst);
+
+        if (_Capacity * sizeof(_Value_type) < 8) {
+            uintptr_t _Alignment_offset = reinterpret_cast<uintptr_t>(_Myfirst) & 0x7;
+            if (_Alignment_offset) {
+                _Alignment_offset = 8 - _Alignment_offset;
+            }
+
+            if (_Capacity * sizeof(_Value_type) > _Alignment_offset) {
+                return reinterpret_cast<const char*>(_Myfirst) + _Alignment_offset;
+            } else {
+                return nullptr;
+            }
+        } else {
+            return reinterpret_cast<void*>((reinterpret_cast<uintptr_t>(_Myfirst) + 7) & ~7);
+        }
+    }
+
+public:
+    constexpr _Asan_annotater(const _Ty& _Target) : _Target_vec(_Target), _Old_last(nullptr), _Aligned_first(nullptr) {}
+
+    _CONSTEXPR20 _Asan_annotater(const _Ty& _Target, _Size_type _New_size) : _Target_vec(_Target) {
+#ifdef __cpp_lib_constexpr_dynamic_alloc
+        if (_STD is_constant_evaluated()) {
+            return;
+        }
+#endif // __cpp_lib_constexpr_dynamic_alloc
+
+        const auto& _My_data     = _Target_vec._Mypair._Myval2;
+        const _Pointer& _Myfirst = _My_data._Myfirst;
+        const _Pointer& _Mylast  = _My_data._Mylast;
+        const _Pointer& _Myend   = _My_data._Myend;
+
+        const _Size_type _Old_capacity = static_cast<_Size_type>(_Myend - _Myfirst);
+
+        if (_New_size > _Old_capacity) {
+            _Old_last = nullptr;
+            return;
+        }
+
+        _Aligned_first = _Get_aligned_start();
+        if (!_Aligned_first) {
+            _Old_last = _Myend;
+        } else {
+            const void* _New_last = static_cast<const void*>(_Myfirst + _New_size);
+            _New_last             = _New_last > _Aligned_first ? _New_last : _Aligned_first;

```

Comment:
I hate pointers,  but the `const` should apply to `_New_last` so why reassign it?